src/internal/exporters/console.rs

```diff
@@ -128,6 +128,12 @@ impl ConsoleWriter {
                 "logfire.level_num" => {
                     if let Value::I64(val) = kv.value {
                         level = Some(val);
+                        if let Some(level) = level {
+                            // Filter out span below the minimum log level
+                            if level < level_to_level_number(self.options.min_log_level) {
+                                return Ok(());
+                            }
+                        }
```

Comment:
It might be worth reordering this to avoid the option check, something like

```rust
if let Value::I64(level_num) = kv.value {
    if level_num < level_to_level_number(self.options.min_log_level) {
        return Ok(());
    }
    level = Some(level_num);
}
```

and similar below in `otel_data_to_writer`.